ui/src/app/analytics/analytics.service.ts

```diff
@@ -117,7 +124,46 @@ export class AnalyticsService {
                       .catch(this.errorHandler.handleError);
   }
 
-  private extractData(response: Response, handleRates: boolean): Page<Counter> {
+  /**
+   * Retrieves all field-value-counters. Will take pagination into account.
+   *
+   * @param detailed If true will request additional counter values from the REST endpoint
+   */
+  private  getAllFieldValueCounters(): Observable<Page<Counter>> {
+    if (!this.counters) {
+      this.counters = new Page<Counter>();
+      this.counters.pageSize = 50;
+    }
+
+    const params = HttpUtils.getPaginationParams(this.counters.pageNumber, this.counters.pageSize);
+    const requestOptionsArgs: RequestOptionsArgs = HttpUtils.getDefaultRequestOptions();
+
+    requestOptionsArgs.search = params;
+    return this.http.get(this.metricsFieldValueCountersUrl, requestOptionsArgs)
+                    .map(response => this.extractData(response, false))
+                    .catch(this.errorHandler.handleError);
+  }
+
+  /**
+   * Retrieves all aggregate counters. Will take pagination into account.
+   *
+   * @param detailed If true will request additional counter values from the REST endpoint
+   */
+  private  getAllAggregateCounters(): Observable<Page<Counter>> {
+    if (!this.counters) {
```

Comment:
It's a bit fishy that this is initialised here and then exact same init block is in `getAllFieldValueCounters` functions. I think we need to find a proper way to set values for a service classes.